Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertDisplayableObject.cs

```diff
@@ -55,45 +55,50 @@ public ApplicationObject DisplayableObjectToNative(Base obj)
     }
   }
 
-  public RevitCategory GetSpeckleObjectCategory(Base @object)
+  public string GetSpeckleObjectBuiltInCategory(Base @object)
   {
-    if (Enum.TryParse<RevitCategory>(@object["category"] as string, out RevitCategory category))
-      return category;
-    else
+    //from 2.16 onwards we're passing the BuiltInCategory on every object
+    if (@object["builtInCategory"] is not null)
+      return @object["builtInCategory"] as string;
+
+    if (RevitCategory.TryParse(@object["category"] as string, out RevitCategory category))
     {
-      switch (@object)
-      {
-        case BE.Beam _:
-        case BE.Brace _:
-        case BE.TeklaStructures.TeklaContourPlate _:
-          return RevitCategory.StructuralFraming;
-        case BE.TeklaStructures.Bolts _:
-          return RevitCategory.StructConnectionBolts;
-        case BE.TeklaStructures.Welds _:
-          return RevitCategory.StructConnectionWelds;
-        case BE.Floor _:
-          return RevitCategory.Floors;
-        case BE.Ceiling _:
-          return RevitCategory.Ceilings;
-        case BE.Column _:
-          return RevitCategory.Columns;
-        case BE.Pipe _:
-          return RevitCategory.PipeSegments;
-        case BE.Rebar _:
-          return RevitCategory.Rebar;
-        case BE.Topography _:
-          return RevitCategory.Topography;
-        case BE.Wall _:
-          return RevitCategory.Walls;
-        case BE.Roof _:
-          return RevitCategory.Roofs;
-        case BE.Duct _:
-          return RevitCategory.DuctSystem;
-        case BE.CableTray _:
-          return RevitCategory.CableTray;
-        default:
-          return RevitCategory.GenericModel;
-      }
+      return Categories.GetBuiltInFromSchemaBuilderCategory(category);
+    }
+
+    switch (@object)
+    {
+      case BE.Beam _:
+      case BE.Brace _:
+      case BE.TeklaStructures.TeklaContourPlate _:
```

Comment:
I have questions, we are matching the object and within the Revit connector we map Beam, Brace and TeklaStructures to the same built in category? It feels a bit brittle, though I accept we've been doing it so far.